File: scripts/pdf_extractor/extract.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extract() {
    try {
        const dataBuffer = fs.readFileSync('../../Manual de Poker/Manual_Grande.pdf');
        const data = await pdf(dataBuffer);
        fs.writeFileSync('manual_grande_text.txt', data.text);
        console.log("Done extracting Manual_Grande.pdf");
    } catch(e) {
        console.error("Error with Manual Grande:", e);
    }

    try {
        const dataBuffer2 = fs.readFileSync('../../Manual de Poker/Curso de Cartomancia con Mazo de Poker parte 1.pdf');
        const data2 = await pdf(dataBuffer2);
        fs.writeFileSync('parte1_text.txt', data2.text);
        console.log("Done extracting parte 1");
    } catch(e) {
        console.error("Error with parte 1:", e);
    }
}

extract();

File: scripts/pdf_extractor/extract.py
from pypdf import PdfReader

def extract_text(pdf_path, output_path, max_pages=15):
    reader = PdfReader(pdf_path)
    text = ""
    for i in range(min(max_pages, len(reader.pages))):
        page = reader.pages[i]
        text += page.extract_text() + "\n"
    with open(output_path, "w", encoding="utf-8") as f:
        f.write(text)

try:
    extract_text("../../Manual de Poker/Manual_Grande.pdf", "manual_grande_index.txt")
    print("Done Manual_Grande")
except Exception as e:
    print(f"Error Manual_Grande: {e}")

try:
    extract_text("../../Manual de Poker/Curso de Cartomancia con Mazo de Poker parte 1.pdf", "parte1_index.txt")
    print("Done parte1")
except Exception as e:
    print(f"Error parte1: {e}")
